src/main/java/frc/robot/Orchestrator.java

```diff
@@ -0,0 +1,287 @@
+package frc.robot;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.ProxyCommand;
+import frc.lib.utils.AllianceFlipUtil;
+import frc.robot.commands.auto.StraightDriveToPose;
+import frc.robot.subsystems.arm.Arm;
+import frc.robot.subsystems.arm.ArmConstants;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.indexer.Indexer;
+import frc.robot.subsystems.indexer.IndexerConstants;
+import frc.robot.subsystems.intake.Intake;
+import frc.robot.subsystems.pixy2.Pixy2;
+import frc.robot.subsystems.robotstate.RobotState;
+import frc.robot.subsystems.shooter.Shooter;
+import java.util.Set;
+import java.util.function.Supplier;
+
+public class Orchestrator {
+  private final Drive drive;
+  private final Intake intake;
+  private final Indexer indexer;
+  private final Shooter shooter;
+  private final Pixy2 pixy2;
+  private final Arm arm;
+  private final Translation2d speaker =
+      AllianceFlipUtil.apply(FieldConstants.Speaker.centerSpeakerOpening.toTranslation2d());
+
+  public Orchestrator(
+      Drive drive, Intake intake, Indexer indexer, Shooter shooter, Pixy2 pixy2, Arm arm) {
+    this.drive = drive;
+    this.intake = intake;
+    this.indexer = indexer;
+    this.shooter = shooter;
+    this.pixy2 = pixy2;
+    this.arm = arm;
+  }
+
+  /**
+   * Drives to one of the notes pre positioned on the field.
+   *
+   * @param targetTranslation The Translation2d for one of the notes on the field during auto.
+   * @return The command for driving to a note during auto.
+   */
+  public Command driveToNote(Translation2d targetTranslation) {
+    Supplier<Rotation2d> targetRotation =
+        () -> targetTranslation.minus(drive.getPose().getTranslation()).getAngle();
+    return Commands.defer(
+        () -> new StraightDriveToPose(new Pose2d(targetTranslation, targetRotation.get()), drive),
```

Comment:
Could you factor out a method to avoid repeating this so much:

```
public Command deferredStraightDriveToPose(Pose2d pose) {
  return Commands.defer(() -> new StraightDriveToPose(pose, drive), Set.of(drive));
}
```

Might be good to also have a `deferredStraightDriveToPose(Rotation2d rotation)` and `deferredStraightDriveToPose(Translation2d translation)`.